src/swap/components/SwapSettingsSlippageInput.tsx

```diff
@@ -0,0 +1,148 @@
+import { type ChangeEvent, useCallback, useEffect, useState } from 'react';
+import { background, border, cn, color, pressable } from '../../styles/theme';
+import type { LifeCycleStatus, SwapSettingsSlippageInputReact } from '../types';
+import { useSwapContext } from './SwapProvider';
+
+function hasMaxSlippage(
+  statusData: any,
+): statusData is { maxSlippage: number } {
+  return statusData?.maxSlippage !== undefined;
+}
+
+function updateMaxSlippage(
+  status: LifeCycleStatus,
+  newMaxSlippage: number,
+): LifeCycleStatus {
+  if (['init', 'amountChange'].includes(status.statusName)) {
+    return {
+      statusName: 'amountChange',
+      statusData: {
+        ...status.statusData,
+        amountFrom: '',
+        amountTo: '',
+        isMissingRequiredField: true,
+        maxSlippage: newMaxSlippage,
+      },
+    };
+  }
+  return status;
+}
+
+export function SwapSettingsSlippageInput({
+  className,
+  defaultSlippage = 3,
+}: SwapSettingsSlippageInputReact) {
+  const { lifeCycleStatus, setLifeCycleStatus } = useSwapContext();
+  const [currentMode, setCurrentMode] = useState<'Auto' | 'Custom'>(() =>
+    ['init', 'amountChange'].includes(lifeCycleStatus.statusName) &&
+    hasMaxSlippage(lifeCycleStatus.statusData) &&
+    lifeCycleStatus.statusData.maxSlippage !== defaultSlippage
+      ? 'Custom'
+      : 'Auto',
+  );
+  const [slippageValue, setSlippageValue] = useState(() =>
+    ['init', 'amountChange'].includes(lifeCycleStatus.statusName) &&
+    hasMaxSlippage(lifeCycleStatus.statusData)
+      ? lifeCycleStatus.statusData.maxSlippage.toString()
+      : defaultSlippage.toString(),
+  );
+
+  useEffect(() => {
+    const newSlippage = Number(slippageValue);
+    if (
+      !Number.isNaN(newSlippage) &&
+      ['init', 'amountChange'].includes(lifeCycleStatus.statusName) &&
```

Comment:
You are overcomplicating this!

That's because you are trying to fit the problem you need to solve in the existing Lifecycle, without noticing, you need a new `stateName` called `slippageChange`.

And this Input the only thing does is:
1. let the user change the slippage
2. when it does, call a new lifecycle event called `slippageChange`.
3. you are done!!!

That's it, simple.

Reply:
Keep your logic simple to understand and simple to write.

If you logic is complex, you are doing something wrong! So go out for a walk, and think how the storytelling of the logic is composed.

And learn how to think both at 1 thousand feet view, 10k feet view, 30k feet view.

If you just work between 1k and 10k, you will lose the bigger picture, and you will write code that gets more complicated over time.

You got this, just go out for a walk and in some cases drawing the logic on a white board or paper helps.

Reply:
I wish I saw these comments earlier!! Awesome, I'm going to wake up tomorrow and knock this out with a fresh brain